### PR TITLE
Angular stub replacements

### PR DESCRIPTION
**Pull request by serhii-lekariev**
This PR substitutes mock feature implementations with their actual counterparts in all components except for `Labels` and `TaskCreationWizard`.

Please note that a Deleted tab still does not work, since enabling it to work would require more `model` module extension (in particular, adding a respective Projection`), which is a matter of further PRs.

It also slightly extends the model by adding a `Status` field to `TaskDetails` message, and making a task explicitly `Open` upon creation.

Now, completed tasks are displayed like so:
completed tasks view.

### CODE REVIEW

File: client-angular/src/app/task-list/active/active-task-item/active-task-item.component.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  ~ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  ~ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+  -->
+<button mat-icon-button class="delete-task-button" (click)="deleteTask(); $event.stopPropagation();">
+    <mat-icon>delete</mat-icon>
+</button>
+<button mat-icon-button class="complete-task-button" (click)="completeTask(); $event.stopPropagation();">
+    <mat-icon>done</mat-icon>
+</button>
+<app-task-display [(task)]="task">
```
Comment (YegorUdovchenko) [inline client-angular/src/app/task-list/active/active-task-item/active-task-item.component.html]: This component was renamed. Please, don`t forget to rename selectors. Also, I would prefer to have both tags on the same line.

File: client-angular/src/app/task-list/active/active-task-item/active-task-item.component.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import {Component, Input} from '@angular/core';
+
+import {TaskItem} from 'generated/main/js/todolist/q/projections_pb';
+import {TaskService} from '../../../task-service/task.service';
+
+/**
+ * The view of a single task in the list.
```
Comment (YegorUdovchenko) [inline client-angular/src/app/task-list/active/active-task-item/active-task-item.component.ts]: In the active tasks list?

File: client-angular/src/app/task-list/active/active-task-item/active-task-item.component.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ */
+@Component({
+  selector: 'app-task-item',
+  templateUrl: './active-task-item.component.html',
+  styleUrls: ['./active-task-item.component.css']
+})
+export class ActiveTaskItemComponent {
```
Comment (YegorUdovchenko) [inline client-angular/src/app/task-list/active/active-task-item/active-task-item.component.ts]: The component selector should correspond to the component class name.

File: client-angular/src/app/task-service/task.service.ts [hunk 4]
```diff
@@ -35,33 +36,29 @@ import {MyListView, TaskItem} from 'generated/main/js/todolist/q/projections_pb'
...
   constructor(private readonly spineWebClient: Client) {
-    this.subscribeToTasks()
-      .then((unsubscribeFn) => this._unsubscribe = unsubscribeFn);
   }
 
   /**
    * Obtains a current view of tasks.
    */
-  get tasks(): TaskItem {
+  get tasks(): BehaviorSubject<TaskItem[]> {
```
Comment (YegorUdovchenko) [inline client-angular/src/app/task-service/task.service.ts]: The `BehaviorSubject` can be modified from outside. It must return plain `Observable`
```
return this._tasks.asObservable();
```

File: client-angular/src/app/task-service/task.service.ts [hunk 5]
```diff
@@ -82,6 +79,32 @@ export class TaskService {
     console.log('Error when sending command to the server: %s', err);
   }
 
+  /**
+   * Deletes the task with the specified ID.
+   *
+   * @param taskId ID of the task to delete.
+   */
+  deleteTask(taskId: string): void {
```
Comment (YegorUdovchenko) [inline client-angular/src/app/task-service/task.service.ts]: Why not use the existing `TaskId` as a parameter? The same for  `completeTask` below.

**Comment by serhii-lekariev** [comment]
@YegorUdovchenko,PTAL again.

File: client-angular/src/app/task-list/deleted/deleted-tasks.component.ts [hunk 6]
```diff
@@ -27,7 +30,14 @@ import {Component} from '@angular/core';
...
-export class DeletedTasksComponent {
+export class DeletedTasksComponent implements OnInit {
+
+  private tasks: TaskItem[];
+
+  constructor(private readonly taskService: TaskService) {
+  }
 
-  constructor() { }
+  ngOnInit(): void {
```
Comment (YegorUdovchenko) [inline client-angular/src/app/task-list/deleted/deleted-tasks.component.ts]: I think this component should use the same approach of querying tasks as the components for active and deleted tasks do. Then you could extract a base class for these components and override only filtering logic.

**Comment by serhii-lekariev** [comment]
@YegorUdovchenko, PTAL again,

File: client-angular/src/app/task-service/task.service.ts [hunk 7]
```diff
@@ -140,4 +172,8 @@ export class TaskService {
         })
     );
   }
+
+  ngOnDestroy(): void {
+    this._unsubscribe();
```
Comment (YegorUdovchenko) [inline client-angular/src/app/task-service/task.service.ts]: `this._unsubscribe` can be `undefined` if the subscription wasn't performed during the service lifecycle. Please add the following check:
```
if (!!this._unsubscribe) {
   this._unsubscribe()
}
```

**Pull request merged**